### PR TITLE
Feature/find package support

### PR DESCRIPTION
**Pull request by tamerharsh**
Implemented find package support for the library, 
Please review.

### CODE REVIEW

File: commandlineparser/CMakeLists.txt [hunk 1]
```diff
@@ -1,2 +1,30 @@
-#CMAKE OF commandlineparser
-add_subdirectory(src)
\ No newline at end of file
+# setting relative paths for the install the directory
+set(include_dest include)
+set(lib_dest lib/${DBUILD_TYPE})
+
+
+#add the library
+add_library(cmd_lib src/Options.cpp src/Parser.cpp include/Parser.h include/Options.h)
```
Comment (inbangsa) [inline commandlineparser/CMakeLists.txt]: make a list of source and header file and pass it here

something like 

file(GLOB helloworld_SRC
    "*.h"
    "*.cpp"
)

File: commandlineparser/CMakeLists.txt [hunk 2]
```diff
@@ -1,2 +1,30 @@
...
+							$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include> 		 
+							$<INSTALL_INTERFACE:${include_dest}> 
+							$<INSTALL_INTERFACE:${lib_dest}> 
+						)
+					   
+#export the target of the library
+install(TARGETS cmd_lib  EXPORT cmd_lib DESTINATION ${lib_dest})
+
+#install the include files
+install(FILES include/Parser.h  DESTINATION ${include_dest})
```
Comment (inbangsa) [inline commandlineparser/CMakeLists.txt]: after getting the list_of_header this two will combine to one statement

File: sample/CMakeLists.txt [hunk 3]
```diff
@@ -1,9 +1,11 @@
+#include files 
 include_directories(${PROJECT_SOURCE_DIR}/commandlineparser/include)
```
Comment (inbangsa) [inline sample/CMakeLists.txt]: Try not including this.. I thing target_link_libraries(SampleApp cmd_lib) should automatically do this.. but I am not sure please check

File: commandlineparser/CMakeLists.txt [hunk 4]
```diff
@@ -1,2 +1,30 @@
...
+install(FILES include/Options.h DESTINATION ${include_dest})
+
+#install the export group
+install(EXPORT cmd_lib DESTINATION ${lib_dest})
+
+#install the <target_name>-config.cmake file 
+install(FILES cmd_lib-config.cmake DESTINATION ${lib_dest})
+
```
Comment (inbangsa) [inline commandlineparser/CMakeLists.txt]: This install commands looking good.. but can be make better with this type of things 

install(TARGETS cmd_lib 
  EXPORT cmd_lib 
  RUNTIME DESTINATION ${bin_dest} COMPONENT cmd_lib 
  LIBRARY DESTINATION ${_lib} COMPONENT cmd_lib 
  ARCHIVE DESTINATION ${lib_dest } COMPONENT cmd_lib 
  PUBLIC_HEADER DESTINATION ${include_dest } COMPONENT cmd_lib 
  )

This automatically install .lib target to lib and .exe target to bin..  you need to define 
set(bin_dest bin)

File: sample/CMakeLists.txt [hunk 5]
```diff
@@ -1,9 +1,11 @@
...
-
+#executable for the sample app
 add_executable(SampleApp sample_user.cpp)
 
-target_link_libraries(SampleApp mylib)
+#linking library cmd_lib
+target_link_libraries(SampleApp cmd_lib)
 
+#install the targets 
 install(TARGETS SampleApp DESTINATION "${PROJECT_SOURCE_DIR}/install/bin")
```
Comment (inbangsa) [inline sample/CMakeLists.txt]: This is wrong install statement you are assuming your install directory 

install(TARGETS SampleApp 
  RUNTIME DESTINATION ${bin_dest} COMPONENT SampleApp 
  )

File: commandlineparser/CMakeLists.txt [hunk 6]
```diff
@@ -1,2 +1,30 @@
-#CMAKE OF commandlineparser
-add_subdirectory(src)
\ No newline at end of file
+# setting relative paths for the install the directory
+set(include_dest include)
```
Comment (inbangsa) [inline commandlineparser/CMakeLists.txt]: This must go in top most cmake as then this can be used by all the sub projects

**Review by inbangsa — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 7]
```diff
@@ -1,5 +1,24 @@
...
+
+# setting relative paths for the install the directory
+set(_include include)
+set(_lib lib)
+set(_bin bin)
+
+
+#adding the directory of sample application and library  
 add_subdirectory(sample)
 add_subdirectory(commandlineparser)
```
Comment (inbangsa) [inline CMakeLists.txt:26]: Sequence of add_subdirectory() matters !!!!

File: CMakeLists.txt [hunk 8]
```diff
@@ -1,5 +1,24 @@
...
+
+if(WIN32)
+  add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+else(WIN32)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-literal-suffix -s")
+endif(WIN32)
+
+# setting relative paths for the install the directory
+set(_include include)
+set(_lib lib)
```
Comment (inbangsa) [inline CMakeLists.txt]: lib bin must take care of  BUILD_TYPES

**Review by inbangsa — CHANGES_REQUESTED** [review]

**Comment by inbangsa** [comment]
Please rebase this

**Comment by inbangsa** [comment]
Pelase resolve conflicts

File: CMakeLists.txt (hunk 7 above)
Comment (inbangsa) [inline CMakeLists.txt:26]: comment not resolved lib must build first

File: commandlineparser/CMakeLists.txt [hunk 9]
```diff
@@ -1,30 +1,35 @@
...
 
-#include lib needed by the library
+#include paths needed by the library to access the include files
 target_include_directories(
 							cmd_lib PUBLIC
 							$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include> 		 
-							$<INSTALL_INTERFACE:${include_dest}> 
-							$<INSTALL_INTERFACE:${lib_dest}> 
+							$<INSTALL_INTERFACE:${_include}> 
+							$<INSTALL_INTERFACE:${_lib}>
```
Comment (inbangsa) [inline commandlineparser/CMakeLists.txt]: Why this _lib

File: README.md [hunk 10]
```diff
@@ -1,2 +1,207 @@
 # CommandLine
 C++ lib for command line parsing 
+
+ifm3d
+=====
+Library and utilities for working with ifm pmd-based 3D ToF Cameras.
```
Comment (inbangsa) [inline README.md]: why ifm3d readme in this commit

File: commandlineparser/CMakeLists.txt [hunk 11]
```diff
@@ -31,5 +31,4 @@ install(FILES ${header_files} DESTINATION ${_include})
 install(EXPORT cmd_lib DESTINATION ${_lib})
 
 #install the <target_name>-config.cmake file 
-install(FILES cmd_lib-config.cmake DESTINATION ${_lib})
-#install(FILES cmd_lib-config.cmake DESTINATION "${_lib}/..")
+install(FILES cmd_lib-config.cmake DESTINATION ${CMAKE_INSTALL_PREFIX})
```
Comment (inbangsa) [inline commandlineparser/CMakeLists.txt]: cant you install this in lib folder

**Review by inbangsa — CHANGES_REQUESTED** [review]

File: commandlineparser/CMakeLists.txt (hunk 11 above)
Comment (tamerharsh) [reply commandlineparser/CMakeLists.txt]: Please clarify lib means a) install/lib b) install/lib/${CONFIG_BUILD_TYPE}.

File: CMakeLists.txt (hunk 7 above)
Comment (tamerharsh) [reply CMakeLists.txt:26]: resolved.

File: commandlineparser/CMakeLists.txt (hunk 9 above)
Comment (tamerharsh) [reply commandlineparser/CMakeLists.txt]: ok ! got it .

File: README.md (hunk 10 above)
Comment (tamerharsh) [reply README.md]: removed., Sorry it got added accidentally.

File: commandlineparser/CMakeLists.txt (hunk 11 above)
Comment (inbangsa) [reply commandlineparser/CMakeLists.txt]: a)

**Review by inbangsa — CHANGES_REQUESTED** [review]

File: commandlineparser/CMakeLists.txt [hunk 12]
```diff
@@ -28,7 +27,12 @@ PUBLIC_HEADER DESTINATION ${_include}
...
 #install the export group
-install(EXPORT cmd_lib DESTINATION ${_lib})
+install(EXPORT cmd_lib
+  FILE
+    cmd_lib.cmake
+  DESTINATION
+  "lib/cmake/cmd_lib"
+)
+
+install(FILES cmd_lib-config.cmake DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/cmake/cmd_lib)
```
Comment (inbangsa) [inline commandlineparser/CMakeLists.txt:37]: {PROJECT_NAME}

**Review by inbangsa — CHANGES_REQUESTED** [review]

File: commandlineparser/CMakeLists.txt (hunk 12 above)
Comment (inbangsa) [reply commandlineparser/CMakeLists.txt:37]: This is not a solution. as This is config file and should not have and Project related variable

**Pull request merged**